packages/browser-sdk/src/lib/react/Grid/index.tsx

```diff
@@ -1,80 +1,129 @@
+                />
+            );
+    }
+}
+
+interface GridVideoCellViewProps {
+    aspectRatio?: number;
+    participant: CellView["client"];
+    isPlaceholder?: boolean;
+    isSubgrid?: boolean;
     render?: () => React.ReactNode;
-    onSetAspectRatio: ({ aspectRatio }: { aspectRatio: number }) => void;
-    onResize?: ({ width, height, stream }: { width: number; height: number; stream: MediaStream }) => void;
-}) {
-    const handleAspectRatioChange = React.useCallback(
-        ({ ar }: { ar: number }) => {
-            if (ar !== cell.aspectRatio) {
-                onSetAspectRatio({ aspectRatio: ar });
-            }
-        },
-        [cell.aspectRatio, onSetAspectRatio],
-    );
+    onSetClientAspectRatio: ({ aspectRatio, clientId }: { aspectRatio: number; clientId: string }) => void;
+}
+
+function GridVideoCellView({ aspectRatio, participant, render, onSetClientAspectRatio }: GridVideoCellViewProps) {
+    const videoEl = React.useRef<WherebyVideoElement>(null);
+
+    const handleResize = React.useCallback(() => {
+        const ar = videoEl.current && videoEl.current.captureAspectRatio();
+
+        if (ar && ar !== aspectRatio && participant?.id) {
+            onSetClientAspectRatio({ aspectRatio: ar, clientId: participant.id });
+        }
+    }, []);
 
     return (
-        <div
-            style={{
-                position: "absolute",
-                width: cell.bounds.width,
-                height: cell.bounds.height,
-                boxSizing: "border-box",
-                top: cell.origin.top,
-                left: cell.origin.left,
-            }}
-        >
+        <div>
             {render ? (
                 render()
-            ) : participant.stream ? (
+            ) : participant?.stream ? (
                 <VideoView
+                    ref={videoEl}
                     stream={participant.stream}
-                    onSetAspectRatio={({ aspectRatio }) => handleAspectRatioChange({ ar: aspectRatio })}
-                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                    onResize={onResize as any}
+                    onVideoResize={handleResize}
+                    style={{
+                        borderRadius: "8px",
+                    }}
                 />
             ) : null}
         </div>
     );
 }
 
 interface GridProps {
-    roomConnection: RoomConnectionRef;
     renderParticipant?: ({
         cell,
         participant,
     }: {
         cell: { clientId: string; bounds: Bounds; origin: Origin };
-        participant: RemoteParticipant | LocalParticipant;
+        participant: CellView["client"];
     }) => React.ReactNode;
     videoGridGap?: number;
 }
 
-function Grid({ roomConnection, renderParticipant, videoGridGap = 0 }: GridProps) {
-    const { remoteParticipants, localParticipant } = roomConnection.state;
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Do we need this comment?

Reply:
yeah it's just because I didn't want to remove this - planning on adding this functionality, just didn't want to make this PR even bigger